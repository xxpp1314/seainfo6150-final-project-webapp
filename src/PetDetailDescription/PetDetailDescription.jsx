import React from 'react';
import PropTypes from "prop-types";
import styles from "./PetDetailDescription.module.css";

const PetDetailDescription= props => {
    return (
        <div className={styles.description}>
            <p className={styles.title}>Price: {props.detail.price}</p><br />
            <p>Category: {props.detail.category}</p><br />
            <p>Age: {props.detail.age}</p><br />
            <p>Gender: {props.detail.gender}</p><br />
            <p>Size: {props.detail.size}</p><br />
            <p>Breed: {props.detail.breed}</p><br />
            <p>Description: {props.detail.description}</p><br />
        </div>
    )
}

PetDetailDescription.propTypes = {
    detail: PropTypes.object.isRequired
};

export default PetDetailDescription;