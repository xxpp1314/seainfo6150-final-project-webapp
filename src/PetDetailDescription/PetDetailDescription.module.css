.description{
    padding: 1%;
    border-radius: 10%;
    width: 100%;
    height: 100%;
    color:rgb(243, 222, 222);
    text-transform: capitalize;
}

.title{
    font-weight: bolder;
    text-transform: capitalize;
}