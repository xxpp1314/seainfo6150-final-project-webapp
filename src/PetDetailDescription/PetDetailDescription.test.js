import React from "react";
import { render } from "@testing-library/react";
import PetDetailDescription from "./PetDetailDescription.jsx";

const petdetails={
    "price":"$300",
    "title":"Russell",
    "category":"dogs",
    "categoryid":"Russell",
    "img":
    {
       "url":"https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/50829956/1/?bust=1619138292&width=1080",
       "height":400,
       "width":500
    },
    "age":"Adult",
    "gender":"Male",
    "size":"Medium",
    "breed":"Terrier Mix",
    "description":["He is a four year old Jack Russell terrier mix. Russell is a great companion pet for adults and families with older children, however he does not do well with other animals so he needs to be an only pet. He is very energetic and loves walks, chewing on his bones, and napping with his owner. He is house trained."]
}

it("renders the PetDetailDescription correctly", () => {
    const { container } = render(
        <PetDetailDescription
            detail={petdetails}
        />
    );
    expect(container).toMatchSnapshot();
});