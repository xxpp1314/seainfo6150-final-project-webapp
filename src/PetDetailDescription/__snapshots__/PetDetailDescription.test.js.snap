// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PetDetailDescription correctly 1`] = `
<div>
  <div
    class="description"
  >
    <p
      class="title"
    >
      Price: 
      $300
    </p>
    <br />
    <p>
      Category: 
      dogs
    </p>
    <br />
    <p>
      Age: 
      Adult
    </p>
    <br />
    <p>
      Gender: 
      Male
    </p>
    <br />
    <p>
      Size: 
      Medium
    </p>
    <br />
    <p>
      Breed: 
      Terrier Mix
    </p>
    <br />
    <p>
      Description: 
      He is a four year old Jack Russell terrier mix. Russell is a great companion pet for adults and families with older children, however he does not do well with other animals so he needs to be an only pet. He is very energetic and loves walks, chewing on his bones, and napping with his owner. He is house trained.
    </p>
    <br />
  </div>
</div>
`;
